docs/spec/auth/vesting.md

```diff
@@ -1,159 +1,138 @@
-SendableCoins(vacc ContinuousVestingAccount, ctx sdk.Context) sdk.Coins:
-    
-    // Coins unlocked by vesting schedule
-    unlockedCoins := ReceivedCoins - SentCoins + OriginalVestingCoins * (Now - StartTime) / (EndTime - StartTime)
-
-    // Must still check for currentCoins constraint since some unlocked coins may have been delegated.
-    currentCoins := vacc.BaseAccount.GetCoins()
+## Vesting Account Implementation
 
-    // min will return sdk.Coins with each denom having the minimum amount from unlockedCoins and currentCoins
-    return min(unlockedCoins, currentCoins)
+Given a vesting account, we may further define the following:
 
+```
+OV = OriginalVesting (constant)
+V = Percentage of OV that is still vesting (derived by OV and the start/end times)
+DV = DelegatedVesting (variable)
+DF = DelegatedFree (variable)
+BC (BaseAccount.Coins) = OV - transferred (can be negative) - delegated (DV + DF)
 ```
 
-The `VestingAccount` interface is used to assert that an account is a vesting account like so:
+### Operations
 
-```go
-vacc, ok := acc.(VestingAccount); ok
-```
+#### Determining Vesting Amount
 
-as well as to calculate the SendableCoins at any given moment.
+To determine the amount of coins that are vested for a given block `B`, the following is performed:
 
-The `ContinuousVestingAccount` struct implements the Vesting account interface. It uses `OriginalVestingCoins`, `ReceivedCoins`, 
-`SentCoins`, `StartTime`, and `EndTime` to calculate how many coins are sendable at any given point. 
-Since the vesting restrictions need to be implemented on a per-module basis, the `ContinuousVestingAccount` implements 
-the `Account` interface exactly like `BaseAccount`. Thus, `ContinuousVestingAccount.GetCoins()` will return the total of 
-both locked coins and unlocked coins currently in the account. Delegated coins are deducted from `Account.GetCoins()`, but do not count against unlocked coins because they are still at stake and will be reinstated (partially if slashed) after waiting the full unbonding period.
+1. Compute `X := B.Time - StartTime`
+2. Compute `Y := EndTime - StartTime`
+3. Compute `V' := OV * (X / Y)`
 
-##### Changes to Keepers/Handler
+Thus, `V := OV - V'`
 
-Since a vesting account should be capable of doing everything but sending with its locked coins, the restriction should be 
-handled at the `bank.Keeper` level. Specifically in methods that are explicitly used for sending like 
-`sendCoins` and `inputOutputCoins`. These methods must check that an account is a vesting account using the check described above.
+#### Transferring/Sending
 
-```go
-if acc is VestingAccount and Now < vestingAccount.EndTime:
-    // Check if amount is less than currently allowed sendable coins
-    if msg.Amount > vestingAccount.SendableCoins(ctx) then fail
-    else:
-        vestingAccount.SentCoins += msg.Amount
+At any given time, a vesting account may transfer: `min((BC + DV) - V, BC)`.
```

Comment:
Let's write this also in English, e.g.

"A vesting account may transfer the minimum of the base account balance and the base account balance plus the number of currently delegated vesting coins less the number of coins vested so far."

Reply:
"...less the number of coins **vesting** so far.", right?

Reply:
Ah yes, I stand corrected.